src/Discovery/test/ClientBase.Test/ConfigurationUrlHelpers.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Configuration;
+using Steeltoe.Discovery.Client;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace Steeltoe.Discovery.ClientBase.Test;
+
+public class ConfigurationUrlHelpers
+{
+    [Fact]
+    public void GetAspNetCoreUrlsHandlesTrailingSemicolon()
+    {
+        var config = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string>() { { "urls", "https://myapp:1234;http://0.0.0.0:1233;http://::1233;http://*:1233;" } }).Build();
```

Comment:
Is the double colon in `http://::1233` intentional? Is that a valid value?

Reply:
It is intentional, but it is not really valid. The format should be [::], but this is the way it has been so I didn't want to change it in 3.2